Lib/typing.py

```diff
@@ -3382,7 +3382,7 @@ def __enter__(self) -> 'TextIO':
 
 class _DeprecatedType(type):
     def __getattribute__(cls, name):
-        if name not in ("__dict__", "__module__") and name in cls.__dict__:
+        if name not in ("__dict__", "__module__", "__doc__") and name in cls.__dict__:
```

Comment:
micro-nit: this could be a set

```suggestion
        if name not in {"__dict__", "__module__", "__doc__"} and name in cls.__dict__:
```